# **Libraries**

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# **Load The Dataset**

In [2]:
df = pd.read_csv(r'C:\Users\Samuel\Documents\Projects\marketing_campaign_analysis\dataset\digital_marketing_campaign_dataset.csv')
df.sample(10)

CustomerID  Age  Gender  Income CampaignChannel   CampaignType  \
1136        9136   31  Female  107160             PPC     Conversion   
6821       14821   68  Female  148670           Email     Conversion   
1554        9554   53  Female   20042    Social Media  Consideration   
4991       12991   18  Female   81377    Social Media      Awareness   
6482       14482   59    Male  116102           Email  Consideration   
2566       10566   28    Male   82714    Social Media     Conversion   
588         8588   27  Female   56894           Email      Retention   
4527       12527   35    Male  135042    Social Media      Retention   
4416       12416   32  Female   78968    Social Media  Consideration   
7171       15171   36  Female   55902             PPC  Consideration   

          AdSpend  ClickThroughRate  ConversionRate  WebsiteVisits  \
1136  7435.833191          0.076232        0.179666             13   
6821   636.606369          0.209231        0.128295              6   
1554  8498.933673          0.072055        0.080039             13   
4991  5125.729341          0.212106        0.086676             27   
6482   436.004314          0.108992        0.141453             39   
2566  1838.278757          0.267666        0.115082             37   
588   2789.501386          0.027655        0.094788             32   
4527   730.718811          0.100879        0.012933             35   
4416  6782.605858          0.024609        0.160185             29   
7171  9168.468535          0.041508        0.146685             44   

      PagesPerVisit  TimeOnSite  SocialShares  EmailOpens  EmailClicks  \
1136       2.064019    3.377667            16           5            9   
6821       3.761489   13.430759            72           9            8   
1554       1.325856    5.868339            59           5            3   
4991       2.428157   12.290597            52          13            7   
6482       7.535145    1.240241            42           6            0   
2566       8.931366    7.168230            84          18            3   
588        2.556227   13.048606            42           5            7   
4527       9.275296    5.863232            22          16            2   
4416       7.117804    5.564638            78           1            1   
7171       2.300022    8.305818             6           0            4   

      PreviousPurchases  LoyaltyPoints AdvertisingPlatform AdvertisingTool  \
1136                  5           4264            IsConfid      ToolConfid   
6821                  6           4430            IsConfid      ToolConfid   
1554                  6           2008            IsConfid      ToolConfid   
4991                  4           3188            IsConfid      ToolConfid   
6482                  5             88            IsConfid      ToolConfid   
2566                  2            187            IsConfid      ToolConfid   
588                   9           3590            IsConfid      ToolConfid   
4527                  4           1951            IsConfid      ToolConfid   
4416                  6           1024            IsConfid      ToolConfid   
7171                  0           2752            IsConfid      ToolConfid   

      Conversion  
1136           1  
6821           1  
1554           1  
4991           1  
6482           1  
2566           1  
588            1  
4527           1  
4416           1  
7171           1

# **Data Exploration**

## Define Data Information Function

In [3]:
def get_data_info(df):
    print("\n\033[1mShape of Dataframe:\033[0m ", df.shape)
    print("\n\033[1mColumns in Dataframe:\033[0m ", df.columns.to_list())
    print("\n\033[1mData Types of Columns:\033[0m\n", df.dtypes)

    print("\n\033[1mInformation About Dataframe:\033[0m")
    df.info()

    print("\n\033[1mNumber of Unique Values in Each Column:\033[0m")
    for col in df.columns:
        print(f"\033[1m{col}\033[0m: {df[col].nunique()}")
    
    print("\n\033[1mNumber of Null Values in Each Column:\033[0m\n", df.isnull().sum())

    print("\n\033[1mNumber of Duplicates Rows:\033[0m ", df.duplicated().sum())

    print("\n\033[1mDescriptive Stats of Dataframe:\033[0m\n", df.describe().transpose())

## Get Info

In [4]:
get_data_info(df)


Shape of Dataframe:  (8000, 20)

Columns in Dataframe:  ['CustomerID', 'Age', 'Gender', 'Income', 'CampaignChannel', 'CampaignType', 'AdSpend', 'ClickThroughRate', 'ConversionRate', 'WebsiteVisits', 'PagesPerVisit', 'TimeOnSite', 'SocialShares', 'EmailOpens', 'EmailClicks', 'PreviousPurchases', 'LoyaltyPoints', 'AdvertisingPlatform', 'AdvertisingTool', 'Conversion']

Data Types of Columns:
 CustomerID               int64
Age                      int64
Gender                  object
Income                   int64
CampaignChannel         object
CampaignType            object
AdSpend                float64
ClickThroughRate       float64
ConversionRate         float64
WebsiteVisits            int64
PagesPerVisit          float64
TimeOnSite             float64
SocialShares             int64
EmailOpens               int64
EmailClicks              int64
PreviousPurchases        int64
LoyaltyPoints            int64
AdvertisingPlatform     object
AdvertisingTool         object
Conversion      

# **Findings**

**What we find about the data**
* The data has 8.000 rows and 20 columns
* There are no duplicate and null values
* There are two confidential columns, AdvertisingPlatform & AdvertisingTool

**Some specific observation about the data**
* **Age**: Average age is around 43 years old, with a range of 18 to 69 years old.
* **Income**: There is a significant spread in income levels due to significant standard deviation with average income around $84.664
* **CTR**:
* **Conversion Rate**:
* **Website Visits**:

In [4]:
df = df.drop(['AdvertisingPlatform', 'AdvertisingTool'], axis=1)

# **Digital Marketing Campaign Success Metrics**

## **1. Click Through Rate (CTR)**
CTR is the number of clicks that ad receives divided by the number of times ad is shown: *clicks ÷ impressions = CTR*.

In [5]:
ctr = (df['ClickThroughRate'].sum() / len(df)) * 100
print(f"Click Through Rate (CTR): {ctr:.2f}%")

Click Through Rate (CTR): 15.48%


## **2. Conversion Rate**

Conversion rate is a percentage used in digital marketing to evaluate performance of website traffic, marketing campaigns and conversions. To calculate a conversion rate, take the number of *conversions divided by the total number of visitors*.

In [6]:
conversion_rate = (df['Conversion'].sum() / len(df)) * 100
print(f"Conversion Rate: {conversion_rate:.2f}%")

Conversion Rate: 87.65%


## **3. Customer Acquisition Cost (CAC)**

Customer Acquisition Cost (CAC) measures how much an organization spends to acquire new customers.

In [7]:
acquired_customers = len(df[df['Conversion'] == 1])
total_cost = df['AdSpend'].sum()
cac = total_cost / acquired_customers

print(f"Customer Acquisition Cost: ${cac:.2f}")

Customer Acquisition Cost: $5705.58


## **4. Marketing Cost by Campaign Channel**

In [8]:
mkt_cost = pd.pivot_table(df,
                          index='CampaignChannel', values='AdSpend',
                          aggfunc='sum', margins=True, margins_name='Total')
mkt_cost = mkt_cost.map(lambda x: '{:,.2f}'.format(x))
mkt_cost

AdSpend
CampaignChannel               
Email             7,871,575.85
PPC               8,199,236.98
Referral          8,653,518.69
SEO               7,740,903.88
Social Media      7,542,323.25
Total            40,007,558.64

## **5. Return on Ad Spend (ROAS)**

In this case, I assume the marketing cost is taken as 5% of the total income. This means, if the marketing cost incurred is 40,007,558.64, then the assumed income obtained is 800,151,173.

In [11]:
total_revenue = (df['AdSpend'].sum()) / 0.05
roas = total_revenue / total_cost
print(f"Return on Ad Spend: {roas:.2f}")

Return on Ad Spend: 20.00


## **6. Return on Investment (ROI)**

**Return on Investment (ROI)** simply compares the profit that resulted from a digital marketing campaign to how much the campaign cost to create and deploy. Ideally, we want as high an ROI as possible. The basic *ROI calculation is: ROI = (Net Profit/Total Cost) x 100*.

In this dataset, we don't have specific revenue column. so I took a different approach by calculating the average revenue per conversion, with the total revenue already defined in the previous row.

In [13]:
total_conversions = len(df[df['Conversion'] == 1])
avg_rev_per_conversion = total_revenue / total_conversions
roi = (avg_rev_per_conversion - total_cost) / total_cost * 100
print(f"Return on Investment: {roi:.2f}%")

Return on Investment: -99.71%


**ROI Analysis**

* **Significant Loss**: A negative ROI of -99.71% means that for every $1 invested, the company loses $0.9971. This indicates that the campaign generated almost no significant revenue, and nearly the entire investment was lost.
* **Major Issue**: This figure indicates a serious problem within the digital marketing campaign, such as a flawed strategy, improper budget allocation, or ineffective execution.
* **Need for Change**: In such situations, it is crucial for the company to immediately reevaluate all aspects of the campaign, including strategy, platforms used, marketing messages, and target audience. Drastic changes or even halting the campaign may be necessary to prevent further losses.

# **Customer Retention Metrics**

## **1. Returning Visitors**

## **2. Email Engagement**

## **3. Previous Purchases**